bobtimus/src/loan.rs

```diff
@@ -304,6 +361,63 @@ mod tests {
     use proptest::proptest;
     use rust_decimal::prelude::FromPrimitive;
     use rust_decimal_macros::dec;
+    use std::str::FromStr;
+
+    #[test]
+    fn test_loan_calculation_and_validation() {
+        let loan_request = LoanRequest {
+            term: 30,
+            principal_amount: LiquidUsdt::from_str_in_dollar("10000").unwrap(),
+            collateralization: dec!(1.4),
+            collateral_amount: Amount::from_btc(0.3675).unwrap().into(),
+
+            // irrelevant for this test
+            collateral_inputs: vec![],
+            borrower_pk: PublicKey::from_str("0218845781f631c48f1c9709e23092067d06837f30aa0cd0544ac887fe91ddd166").unwrap(),
+            borrower_address: Address::from_str("el1qq0zel5lg55nvhv9kkrq8gme8hnvp0lemuzcmu086dn2m8laxjgkewkhqnh8vxdnlp4cejs3925j0gu9n9krdgmqm89vku0kc8").unwrap()
+        };
+
+        let loan_offer = LoanOffer {
+            min_principal: LiquidUsdt::from_str_in_dollar("1000").unwrap(),
+            max_principal: LiquidUsdt::from_str_in_dollar("10000").unwrap(),
+            max_ltv: dec!(0.75),
+            base_interest_rate: dec!(0.05),
+
+            // TODO: Specify terms once term validation is implemented
+            terms: vec![],
+            collateralizations: vec![],
+
+            // irrelevant for this test
+            rate: Rate {
+                ask: Default::default(),
+                bid: Default::default(),
```

Comment:
It might still be useful to state it here so that one does not need to do the math on how the collateral_amount was chosen. 

As a reference, that's what I came up with: 

```
// collateral_amount / collateralization_percent = value of principal_amount

0.3675 / 1.4 = 0.2625

// so

0.2625 BTC = 10,000

// which means, 

10,000 * 1/0.2625 = BTC price 

rate = 38095 USDT/BTC

```

now with this and the given `max_ltv` of `0.75` I get: 

```
initial_price * LTV = liquidation_threshold
38095 * 0.75 = 28571.25
```

I ignored the 0.05% interest rate but the value seems to be close to what you have here.